### DISCUSSION TITLE
Terminal fails to redraw most of the time when switching to Blink on 9.7" iPad Pro

### DISCUSSION THREAD

**Discussion by jgrg** [blinksh/blink#1745] [Support] [answered]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

Blink v16.1.3 on iPadOS 16.3.1 on 9.7" iPad Pro

Same issue was present in the previous release of Blink.

### Describe the bug

Screen is not redrawing when I switch back to Blink. It doesn't fail every time, but probably more often than not. The visible area of the terminal above the keyboard flashes white, then black, then my chosen grey background colour, but the text is not drawn. If I force quit and relaunch Blink, the screen redraws correctly.

I notice that when the screen does redraw correctly, a little info panel appears in the centre of the screen with the text "blink" and the terminal size in columns and rows. This info panel doesn't appear when it fails to redraw.

Image of screen when redraw fails:

![IMG_0160](https://user-images.githubusercontent.com/6400382/226099477-4105ad8e-298b-45a6-8b02-759f54067c3b.PNG)

**Comment by carloscabanero** [comment, marked as answer]
This may happen some times with a blank Blink Shell. Just pressing enter will redraw the shell prompt again. We will be making changes to the underlying structure of the shell that should get rid of this issue, but will not fix the bug as-is.

**Comment by jgrg** [comment]
This now appears to be fixed in version 16.1.7
I haven't seen it happen in this version, and the screen doesn't flash when switching to Blink either. (Or maybe it does still flash, but I don't notice because I've switched to a light background.)
